hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/HoodieSparkClusteringClient.java

```diff
@@ -47,8 +52,11 @@ public HoodieSparkClusteringClient(
   public void cluster(HoodieInstant instant) throws IOException {
     LOG.info("Executing clustering instance " + instant);
     SparkRDDWriteClient<T> writeClient = (SparkRDDWriteClient<T>) clusteringClient;
-    JavaRDD<WriteStatus> res = writeClient.cluster(instant.getTimestamp(), true).getWriteStatuses();
-    if (res != null && res.collect().stream().anyMatch(WriteStatus::hasErrors)) {
+    Option<HoodieCommitMetadata> commitMetadata = writeClient.cluster(instant.getTimestamp(), true).getCommitMetadata();
+    List<HoodieWriteStat> writeStats = commitMetadata.get().getPartitionToWriteStats().entrySet().stream().flatMap(e ->
+            e.getValue().stream()).collect(Collectors.toList());
```

Comment:
minor: Can we just use Stream<HoodieWriteStat>? Do you really need to collect all WriteStats to a list? We are converting it back to stream in next line.

Reply:
Sure thing! Thanks a lot for your review. changed :)